### PR TITLE
use nvidia graphics automatically on laptops with optimus

### PR DESCRIPTION
**Pull request by BreadFish64**
https://stackoverflow.com/a/39047129
basically copied this^
I held off on AMD because I have no way to test that, and AMD GPUs sometimes have more issues than Intel integrated graphics in the current PICA->GLSL implementation.
(this can of course be overridden by custom control panel profiles)

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3528)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -54,6 +54,12 @@
 Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin);
 #endif
 
+#ifdef _WIN32
+extern "C" {
+__declspec(dllexport) unsigned long NvOptimusEnablement = 0x00000001;
```
Comment (Subv) [inline src/citra_qt/main.cpp]: Won't this break in the mingw builds? Or does gcc also support __declspec(dllexport)?

Comment (BreadFish64) [reply src/citra_qt/main.cpp]: it appears to work fine

File: src/citra/citra.cpp [hunk 2]
```diff
@@ -40,6 +40,12 @@
 #include "core/settings.h"
 #include "network/network.h"
 
+#ifdef _WIN32
+extern "C" {
+__declspec(dllexport) unsigned long NvOptimusEnablement = 0x00000001;
```
Comment (degasus) [inline src/citra/citra.cpp]: Please add comments about what this flag does and why we want to have it. It is easier to read a comment than to use git blame ;)

File: src/citra_qt/main.cpp (hunk 1 above)
Comment (wwylele) [reply src/citra_qt/main.cpp]: https://gcc.gnu.org/onlinedocs/gcc/Microsoft-Windows-Function-Attributes.html#Microsoft-Windows-Function-Attributes
gcc supports this on windows

**Pull request merged**